Sockets/Sockets.Implementation.NET/TcpSocketClient.cs

```diff
@@ -36,11 +38,41 @@ internal TcpSocketClient(TcpClient backingClient)
         /// </summary>
         /// <param name="address">The address of the endpoint to connect to.</param>
         /// <param name="port">The port of the endpoint to connect to.</param>
-        public async Task ConnectAsync(string address, int port)
+        /// <param name="secure">Is this connection secure?</param>
+        public async Task ConnectAsync(string address, int port, bool secure = false)
         {
             await _backingTcpClient.ConnectAsync(address, port);
+            if (!secure)
+            {
+                _secureStream = new SslStream(_backingTcpClient.GetStream(), true, (sender, cert, chain, sslPolicy) => ServerValidationCallback(sender,cert,chain,sslPolicy));
+                _secureStream.AuthenticateAsClient(address, null, System.Security.Authentication.SslProtocols.Tls, false);
+            }            
         }
 
+        #region Secure Sockets Details
+        
+        private bool ServerValidationCallback (object sender, X509Certificate certificate, X509Chain chain, SslPolicyErrors sslPolicyErrors)
+        {
+            switch (sslPolicyErrors)
```

Comment:
Here for errors I would prefer not to use Console.WriteLine, as the user can't prevent the output. I would prefer to throw an exception and allow the user to catch it if they want, but I think that returning `false` here will result in `AuthenticateAsClient` throwing an exception anyway, is that correct?